Add pre-commit hook to check if the current branch needs rebase

bernhardkaindl opened the pull request:
Add pre-commit hook to check if the current branch needs rebase
- GitHub runs CI on the PR branch rebased to the target branch if possible.
- To avoid hard side effects from working on the branch that is behind the target, warn about it.

GeraldEV commented on .github/workflows/check_branch-needs-rebase.py:
```diff
@@ -0,0 +1,83 @@
+    )
+    if err:
+        target = "origin/master"
+        # Get the default branch name of the remote repository
```
Should this comment be after the else line?

bernhardkaindl replied:
Thanks for the review! Correct, this was the wrong location!

Moved it and added the correct comment at that place!

bernhardkaindl commented on .github/workflows/check_branch-needs-rebase.py:
```diff
@@ -0,0 +1,83 @@
+def check_if_local_branch_is_behind_target(current_branch, target="origin/master"):
+    """Check if the current branch is behind the target branch."""
+
+    log(f"The current branch {current_branch} does not have an upstream branch")
```
Thanks for the review! Correct!

I updated this and fixed it, added missing comments as well!

bernhardkaindl commented:
@GeraldEV Thanks for the review!

Thanks for hinting me at the log() call and the comment!

I updated both, fixing them and moved the comment and added missing comments as well!

ashwin9390 commented on .github/workflows/check_branch-needs-rebase.py:
pylance throws annotation error during runtime for list and tuple ,should you consider using quotes for it ?.

bernhardkaindl replied:
I guess you refer to:
```py
def run(command: list[str], check=False) -> tuple[int, str]:
```
This these annotations are fine for Python3 code, and VS Code/Pylance should only warn a warning about it if you selected Python 2.7 as the active Python interpreter in VS Code: Don't do that normally. It's better if you select a Python3 version as the active interpreter.

Pyright (which is the static checker in Pylance) has no issues:
```py
pyright .github/workflows/check_branch-needs-rebase.py
0 errors, 0 warnings, 0 informations 
```
